### ISSUE TITLE
[docs] New API format

### ISSUE THREAD

**Issue by PupoSDC** [mui/mui-x#9847]
### Duplicates

- [X] I have searched the existing issues

### Related page

https://mui.com/material-ui/api/button/

### Kind of issue

Other

### Issue description

This past week the API docs format has changed dramatically. I couldn't find a discussion open on it, so I'd like to start one. 

I can't exactly point my finger on "why" but it's just so hard to get information from it. It's just too spread out and i I find it very difficult to have a nice helicopter view of a component's API now, to the point where Im finding myself going to the source code to get a more compact view of the docs.

To compare, I plucked a picture from the old mui v4 docs which are still online and the new docs: 

<img width="1507" alt="image" src="https://github.com/mui/mui-x/assets/9625304/e096d2d5-1182-4a09-a2db-d9a822541e54">

<img width="1512" alt="image" src="https://github.com/mui/mui-x/assets/9625304/afb8dc9b-aaec-454d-b483-631d558004c9">

To compare In the old format I can overview 8 props at the same time. In the new docs it's down to 4. But it's not just that... it's just much harder to extract information from it.  99% of the time I only care about the `name` and `description` column... It's just so easy to filter out mentally the middle columns when skimming through a table.... meanwhile a List does not give that benefit. Likewise... getting a fast overview of which components have `default` values is trivial on the old table... now it's just not possible.  

In summary it feels like the docs moved from the nice easy to read format you find in tools like [`Storybook controls`](https://storybook.js.org/docs/react/api/doc-block-controls) to the [PTSD inducing style of Java docs](https://docs.oracle.com/javase/8/docs/api/).

I hope you guys give this design a second thought, unless of course, I'm the only dissident voice.

Keep up the great work!

**Comment by PupoSDC** [comment]
Closed in favour of https://github.com/mui/material-ui/issues/38245
Sorry for the noise.
